### PR TITLE
Added RangeDelete to kv cli

### PR DESCRIPTION
**Pull request by ajayaa**

### CODE REVIEW

File: cli/kv.go [hunk 1]
```diff
@@ -286,6 +286,33 @@ func runReverseScan(cmd *cobra.Command, args []string) {
 	showResult(rows)
 }
 
+var rangeDelCmd = &cobra.Command{
+	Use:   "rangedel [options] <startKey> <endKey>",
```
Comment (tamird) [inline cli/kv.go]: There's a TODO at the top of this file mentioning DeleteRange - could you remove that bit?

File: cli/kv.go [hunk 2]
```diff
@@ -286,6 +286,33 @@ func runReverseScan(cmd *cobra.Command, args []string) {
...
+func runRangeDel(cmd *cobra.Command, args []string) {
+	if len(args) != 2 {
+		cmd.Usage()
+		return
+	}
+	kvDB := makeDBClient()
+	if kvDB == nil {
+		return
+	}
+	startKey := proto.Key(unquoteArg(args[0], false))
```
Comment (tamird) [inline cli/kv.go]: let's change these `false`s to `true`s; we don't want to allow system keys to be deleted. (see style in #2103)

**Comment by tamird** [comment]
Thanks for the PR! could you add an example to `cli_test.go`?

**Comment by tschottdorf** [comment]
why not `./kv del --range a b [ c d [ ... ] ]` instead of new command?

**Comment by ajayaa** [comment]
I was just reading the source file and found an easy enough TODO. So wrote it and raise a PR. I didn't know that you could delete ranges using --range flag in delete. Seems like rangedel is not needed after all. I would just remove the TODO flag in the file.

@tamird @tschottdorf thoughts?

**Comment by tamird** [comment]
@ajayaa I think @tschottdorf was suggesting that you add the `--range` flag to the existing `del` command.

**Comment by ajayaa** [comment]
@tamird ahh, okay! That is doable. :+1:

**Comment by tschottdorf** [comment]
sorry, that's what I ment. Does that sound like a good idea?

**Comment by ajayaa** [comment]
sounds good. :)

File: cli/kv.go [hunk 3]
```diff
@@ -201,13 +205,26 @@ func runDel(cmd *cobra.Command, args []string) {
...
-		unquoted = append(unquoted, unquoteArg(args[0], i%2 == 0 /* disallow system keys */))
+		unquoted = append(unquoted, unquoteArg(args[i], true /* disallow system keys */))
 	}
 
 	kvDB := makeDBClient()
 	if kvDB == nil {
 		return
 	}
+
+	if delRange == true {
```
Comment (petermattis) [inline cli/kv.go]: `s/delRange == true/delRange/g`.

File: cli/kv.go [hunk 4]
```diff
@@ -187,7 +189,9 @@ var delCmd = &cobra.Command{
 	Use:   "del [options] <key> [<key2>...]",
 	Short: "deletes the value for a key",
 	Long: `
-Deletes the value for one or more keys.
+Deletes the value for one or more keys. Only two keys are expected when
+--range flag is used. If the --range flag is used then all the keys in
```
Comment (petermattis) [inline cli/kv.go]: `s/used/specified/g` (for both instances of "used").

File: cli/kv.go [hunk 5]
```diff
@@ -187,7 +189,9 @@ var delCmd = &cobra.Command{
...
+range of <key1> and <key2> is deleted including <key1> and excluding <key2>.
```
Comment (petermattis) [inline cli/kv.go]: `s/range/the range/g`

I would have used the standard notation of open and closed ranges. Something like: `the range [<key1>, <key2>)`, but perhaps that is too terse.

**Comment by petermattis** [comment]
LGTM, though it looks like you'll need to rebase.

File: cli/flags.go [hunk 6]
```diff
@@ -228,6 +228,13 @@ func initFlags(ctx *server.Context) {
 		f := cmd.Flags()
 		f.Int64Var(&maxResults, "max-results", defaultMaxResults, flagUsage["max-results"])
 	}
+
+	// range flag for delete.
+	for _, cmd := range []*cobra.Command{delCmd} {
```
Comment (tamird) [inline cli/flags.go]: iterating over just one command is probably not necessary.
```go
{
  f := delCmd.Flags()
  f.BoolVar(&delRange, "range", false, flagUsage["range"])
}
```

You also need to add something to `flagUsage` under the key "range" though I think you'll want a more descriptive name than "range".

File: cli/flags.go [hunk 7]
```diff
@@ -231,6 +231,10 @@ func initFlags(ctx *server.Context) {
 		f := cmd.Flags()
 		f.Int64Var(&maxResults, "max-results", defaultMaxResults, flagUsage["max-results"])
 	}
+
+	// range flag for delete.
+	f := delCmd.Flags()
```
Comment (tamird) [inline cli/flags.go]: can you scope this like we do elsewhere? i.e.
```
{ 
  f := ...
  f.BoolVar(...)
}
```

File: cli/kv.go [hunk 8]
```diff
@@ -212,6 +216,19 @@ func runDel(cmd *cobra.Command, args []string) {
...
 		return
 	}
+
+	if delRange {
+		if len(keys) != 2 {
+			cmd.Usage()
+			return
+		}
+		if err := kvDB.DelRange(keys[0], keys[1]); err != nil {
+			fmt.Fprintf(osStderr, "del with --range failed: %s\n", err)
```
Comment (tamird) [inline cli/kv.go]: "range delete failed"

File: cli/kv.go [hunk 9]
```diff
@@ -212,6 +216,19 @@ func runDel(cmd *cobra.Command, args []string) {
 	if kvDB == nil {
 		return
 	}
+
+	if delRange {
+		if len(keys) != 2 {
```
Comment (tamird) [inline cli/kv.go]: this check should happen before we create a client

**Pull request closed without merging**